soccer/radio/USBRadio.cpp

```diff
@@ -174,65 +180,128 @@ void USBRadio::send(Packet::RadioTx& packet) {
     static const float Radians_Per_Tick = 0.026f * M_PI / (0.0812f * 3240.0f);
 
-    rtp::header_data* header = (rtp::header_data*)forward_packet;
-    header->port = rtp::Port::CONTROL;
+    rtp::Header* header = (rtp::Header*)forward_packet;
+    header->port = rtp::PortType::CONTROL;
     header->address = rtp::BROADCAST_ADDRESS;
-    header->type = rtp::header_data::Type::Control;
+    header->type = rtp::MessageType::CONTROL;
 
     // Build a forward packet
     for (int slot = 0; slot < 6; ++slot) {
         // Calculate the offset into the @forward_packet for this robot's
         // control message and cast it to a ControlMessage pointer for easy
         // access
         size_t offset =
-            sizeof(rtp::header_data) + slot * sizeof(rtp::ControlMessage);
-        rtp::ControlMessage* msg =
-            (rtp::ControlMessage*)(forward_packet + offset);
+            sizeof(rtp::Header) + slot * sizeof(rtp::RobotTxMessage);
+        rtp::RobotTxMessage* msg =
+            (rtp::RobotTxMessage*)(forward_packet + offset);
 
         if (slot < packet.robots_size()) {
             const Packet::Control& robot = packet.robots(slot).control();
 
             msg->uid = packet.robots(slot).uid();
+            msg->messageType = rtp::RobotTxMessage::ControlMessageType;
 
-            msg->bodyX =
-                robot.xvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR;
-            msg->bodyY =
-                robot.yvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR;
-            msg->bodyW =
-                robot.avelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR;
+            auto& controlMessage = msg->message.controlMessage;
 
-            msg->dribbler =
-                max(0, min(255, static_cast<uint16_t>(robot.dvelocity()) * 2));
+            controlMessage.bodyX = static_cast<int16_t>(
+                robot.xvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
+            controlMessage.bodyY = static_cast<int16_t>(
+                robot.yvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
+            controlMessage.bodyW = static_cast<int16_t>(
+                robot.avelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
 
-            msg->kickStrength = robot.kcstrength();
+            controlMessage.dribbler =
+                clamp(static_cast<uint16_t>(robot.dvelocity()) * 2, 0, 255);
 
-            msg->shootMode = robot.shootmode();
-            msg->triggerMode = robot.triggermode();
-            msg->song = robot.song();
+            controlMessage.kickStrength = robot.kcstrength();
+            controlMessage.shootMode = robot.shootmode();
+            controlMessage.triggerMode = robot.triggermode();
+            controlMessage.song = robot.song();
         } else {
             // empty slot
             msg->uid = rtp::INVALID_ROBOT_UID;
         }
     }
 
+    // COMMENTED OUT CODE FOR DEBUG PACKETS.
+    //    int numRobotTXMessages = packet.robots_size();
```

Comment:
This needs to be verified and cleaned up, but I'm fine with leaving that for another PR.